transport.go

```diff
@@ -13,9 +13,29 @@ type WrapErrorFunc func(error) interface{}
 type Transporter interface {
 	getDispatcher() (dispatcher, error)
 	getReceiver() (receiver, error)
+
+	// Done returns a channel that's closed when incoming packets
+	// have finished processing, either due to an error or the
+	// underlying connection being closed. Successive calls to
+	// Done return the same value. This is safe to call before
+	// incoming packets have started processing (i.e., before
+	// Run() or RunAsync() has been called).
+	Done() <-chan struct{}
```

Comment:
I went with Done() instead of GetCloseChannel() or something since it's not necessarily when the connection is closed that stopCh is closed (could be a framing error, etc.)